### PR TITLE
Add Button Action (citation, comment, download, share)

### PR DESCRIPTION
**Pull request by Cosmin-Daniel**

### CODE REVIEW

**Comment by Cosmin-Daniel** [comment]
Closing in favour of: #927

File: tests/src/ViewModel/ButtonActionTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * @test
+     */
+    public function it_has_data()
+    {
+        $data = [
+            'text' => 'Button action (citation)',
+            'classes' => 'button--default button--action icon icon-citation',
+            'id' => 'button-action-citation',
+            'path' => '#citation'
```
Comment (nlisgo) [inline tests/src/ViewModel/ButtonActionTest.php:19]: nitpick: comma at end of line

File: src/ViewModel/Button.php [hunk 2]
```diff
@@ -124,6 +129,32 @@ public static function link(
         return $button;
     }
 
+    public static function action(
+        string $text,
```
Comment (nlisgo) [inline src/ViewModel/Button.php:133]: add path and id as parameters. id should be optional.

Comment (Cosmin-Daniel) [reply src/ViewModel/Button.php:133]: Parameter path and id (optional) have been added.

File: tests/src/ViewModel/ButtonActionTest.php (hunk 1 above)
Comment (Cosmin-Daniel) [reply tests/src/ViewModel/ButtonActionTest.php]: Comma has been added. Thank you!

**Review by nlisgo — APPROVED** [review]

**Pull request merged**